qutebrowser/mainwindow/tabwidget.py

```diff
@@ -472,9 +504,31 @@ def tabSizeHint(self, index):
             # get scroll buttons as soon as needed.
             size = minimum_size
         else:
+            tab_width_pinned_conf = config.get('tabs', 'pinned-width')
+
+            try:
+                pinned = self.tab_data(index, 'pinned')
+            except KeyError:
+                pinned = False
+
+            if pinned:
+                size = QSize(tab_width_pinned_conf, height)
+                qtutils.ensure_valid(size)
+                return size
+
             # If we *do* have enough space, tabs should occupy the whole window
-            # width.
-            width = self.width() / self.count()
+            # width. If there is pinned tabs their size will be substracted
+            # from the total window width.
+            # During shutdown the self.count goes down,
+            # but the self.pinned_count not - this generates some odd behavior.
+            # To avoid this we compare self.count against self.pinned_count.
+            if self.pinned_count > 0 and self.count() > self.pinned_count:
```

Comment:
In theory, no. The only expected case in here is `pinned_count == 0`, but I'd rather leave it as is, because if there is a problem counting tabs (and we get a negative value), we would fall back to a situation where we assume we have ~0 pinned tabs. I can switch it over to being `not pinned_count` if you want though :P.